### PR TITLE
feat(linter): no-await-expression-member for eslint-plugin-unicorn

### PR DESCRIPTION
**Pull request by Ken-HH24**
Try to implement `no-await-expression-member` for #684

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
Can you add typescript test cases as well? 
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/no-await-expression-member.mjs#L48-L61

File: crates/oxc_linter/src/rules/unicorn/no_await_expression_member.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-await-expression-member): Disallow member access from await expression")]
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/unicorn/no_await_expression_member.rs:12]: `eslint-plugin-unicorn`?

**Comment by Ken-HH24** [comment]
> Can you add typescript test cases as well? https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/no-await-expression-member.mjs#L48-L61

I have added typescript test cases for the rule.

File: crates/oxc_linter/src/rules/unicorn/no_await_expression_member.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// const secondElement = (await getArray())[1];
+    ///
+    /// // Good
+    /// const [, secondElement] = await getArray();
+    /// ```
+    NoAwaitExpressionMember,
+    correctness
```
[the remark this answers is not in the document]
Comment (Ken-HH24) [reply crates/oxc_linter/src/rules/unicorn/no_await_expression_member.rs]: Thank you for the correction

**Review by IWANABETHATGUY — APPROVED** [review]

**Review by Boshen — APPROVED** [review]

**Pull request merged**